syntax = "proto3";

package sekai.run_analysis;

message Status {
  optional int32 code = 1;
  optional string msg = 2;
}

message DeckCard {
  optional int32 card_id      = 1;
  optional int32 master_rank  = 2;
  optional bool trained_state = 3;
}

message AnalyzeTeamRequest {
  optional int32 event_id   = 1;
  repeated DeckCard cards   = 2;
  optional int32 team_power = 3;
}

message AnalyzeTeamResponse {
  optional Status status = 1;

  // The main team event bonus.
  //
  // Note that for WL events, this does not include the support team bonus.
  optional float event_bonus = 2;

  // Potential skill range for this team.
  message SkillDetails {
    repeated int32 skill_lower_bounds      = 1;
    repeated int32 skill_upper_bounds      = 2;
    optional float skill_value_lower_bound = 3;
    optional float skill_value_upper_bound = 4;
  }
  optional SkillDetails skill_details = 3;
}
